pkg/utils/apiutil/serverapi/middleware.go

```diff
@@ -122,17 +122,19 @@ func (h *redirector) matchMicroServiceRedirectRules(r *http.Request) (bool, stri
 	// It will be helpful when matching the redirect rules "schedulers" or "schedulers/{name}"
 	r.URL.Path = strings.TrimRight(r.URL.Path, "/")
 	for _, rule := range h.microserviceRedirectRules {
-		if strings.HasPrefix(r.URL.Path, rule.matchPath) && slice.Contains(rule.matchMethods, r.Method) {
+		if strings.HasPrefix(r.URL.Path, rule.matchPath) &&
+			slice.Contains(rule.matchMethods, r.Method) {
 			if rule.filter != nil && !rule.filter(r) {
 				continue
 			}
-			origin := r.URL.Path
+			// we check the service primary addr here, so no need to check independently again.
```

Comment:
Is it right? cc @rleungx

Reply:
Yes, because the check service is used to manage service dynamically and tso is not supported now.